testutils/infra/container_manager/docker_compose_manager.py

```diff
@@ -232,94 +231,99 @@ def get_mender_conductor(self):
     def teardown(self):
         self.stop_docker_compose()
 
 class DockerComposeLegacyClientSetup(DockerComposeNamespace):
     def __init__(self, name, ):
-        DockerComposeNamespace.__init__(self, name)
+        DockerComposeNamespace.__init__(self, name, self.LEGACY_CLIENT_FILES)
     def setup(self):
-        self.docker_compose_cmd("up -d", use_common_files=False, file_list=self.BASE_FILES+self.LEGACY_CLIENT_FILES)
+        self._docker_compose_cmd("up -d")
     def teardown(self):
         self.stop_docker_compose()
 
 class DockerComposeSignedArtifactClientSetup(DockerComposeNamespace):
     def __init__(self, name, ):
-        DockerComposeNamespace.__init__(self, name)
+        DockerComposeNamespace.__init__(self, name, self.QEMU_CLIENT_FILES+self.SIGNED_ARTIFACT_CLIENT_FILES)
     def setup(self):
-        self.docker_compose_cmd("up -d", use_common_files=True, file_list=self.SIGNED_ARTIFACT_CLIENT_FILES)
+        self._docker_compose_cmd("up -d")
     def teardown(self):
         self.stop_docker_compose()
 
 class DockerComposeShortLivedTokenSetup(DockerComposeNamespace):
     def __init__(self, name, ):
-        DockerComposeNamespace.__init__(self, name)
+        DockerComposeNamespace.__init__(self, name, self.QEMU_CLIENT_FILES+self.SHORT_LIVED_TOKEN_FILES)
     def setup(self):
-        self.docker_compose_cmd("up -d", use_common_files=True, file_list=self.SHORT_LIVED_TOKEN_FILES)
+        self._docker_compose_cmd("up -d")
     def teardown(self):
         self.stop_docker_compose()
 
 class DockerComposeFailoverServerSetup(DockerComposeNamespace):
     def __init__(self, name, ):
-        DockerComposeNamespace.__init__(self, name)
+        DockerComposeNamespace.__init__(self, name, self.QEMU_CLIENT_FILES+self.FAILOVER_SERVER_FILES)
     def setup(self):
-        self.docker_compose_cmd("up -d", use_common_files=True, file_list=self.FAILOVER_SERVER_FILES)
+        self._docker_compose_cmd("up -d")
     def teardown(self):
         self.stop_docker_compose()
 
 class DockerComposeEnterpriseSetup(DockerComposeNamespace):
     def __init__(self, name, num_clients=0):
         self.num_clients = num_clients
-        DockerComposeNamespace.__init__(self, name)
-    def setup(self):
         if self.num_clients > 0:
-            raise NotImplementedError("Clients not implemented for Enterprise setup")
+            raise NotImplementedError("Clients not implemented on setup time, use new_tenant_client")
         else:
-            self.docker_compose_cmd("up -d", use_common_files=False, file_list=self.ENTERPRISE_FILES)
-            self.wait_for_containers(15, defined_in=self.ENTERPRISE_FILES)
+            DockerComposeNamespace.__init__(self, name, self.ENTERPRISE_FILES)
+    def setup(self, recreate=True, env=None):
+        cmd = "up -d"
+        if not recreate:
+            cmd += " --no-recreate"
+        self._docker_compose_cmd(cmd, env=env)
+        self.wait_for_containers(15)
```

Comment:
Can we factor out this magic number (15) to a global variable?
When I've added the workflows service I had to increment it and this wasn't something obvious.

Reply:
Thanks for the suggestion.